client/my-sites/plugins/plugins-list/index.jsx

```diff
@@ -131,12 +131,12 @@ export default React.createClass( {
 		const bulkManagement = ! this.state.bulkManagement;
 
 		if ( bulkManagement ) {
+			this.setBulkSelectionState( this.props.plugins, bulkManagement );
 			this.setState( { bulkManagement } );
 			return this.recordEvent( 'Clicked Manage' );
 		}
 
-		// Unselect all plugins.
-		this.setState( { selectedPlugins: {}, bulkManagement } );
```

Comment:
'Edit all' implies to me that all the plugins will be checked for edits. I think preserving state across toggles would require a new button label. Something that would convey that user is no longer in the initial state. 

I could imagine the button label turning to something like 'Continue Editing' if the user opens the bulk management, un-selects some plugins and then closes bulk management. But I'm weary of that type of UI change being more unexpected than having all the plugins selected when clicking edit all.

I can mock that up to see how it looks